lib/DDG/Spice/IsItUp.pm

```diff
@@ -3,28 +3,35 @@ package DDG::Spice::IsItUp;
 
 use strict;
 use DDG::Spice;
-use DDG::Util::SpiceConstants;
+use Domain::PublicSuffix;
+use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
+use Net::Domain::TLD qw(tld_exists);
+use Data::Validate::IP qw(is_ipv4);
 
-triggers query_lc => qr/^((?:is\s|))(?:https?:\/\/)?([0-9a-z\-]+(?:\.[0-9a-z\-]+)*?)(?:(\.[a-z]{2,4})|)\s(?:up|down|working|online|status)\?*$/i;
+triggers query_lc => qr/^((?:is\s|))(?:https?:\/\/)?([\p{Alnum}\-]+(?:\.[\p{Alnum}\-]+)*?)(?:(\.\pL{2,})|)\s(?:up|down|working|online|status)\?*$/i;
 
 spice to => 'https://isitup.org/$1.json?callback={{callback}}';
 
 spice proxy_cache_valid => "418 1d";
 
-my $regex_domain = qr/\.(@{[ DDG::Util::SpiceConstants::TLD_REGEX  ]})$/;
-my $regex_ipv4 = qr/^(?:\d{1,3}\.){3}\d{1,3}$/;
-
 handle matches => sub {
+
+    my ($domain, $ascii);
+    my $publicSuffix = Domain::PublicSuffix->new();
+
     if ($_[2]) {
-        my $root_url = $_[1];
-        my $domain = $_[2];
```

Comment:
Any reason you removed these variables? IMO using named variables makes the code easier to read/follow.